src/k8s/pkg/component/component.go

```diff
@@ -44,52 +54,68 @@ type Component struct {
 	Status bool
 }
 
+// InitializeHelmClientConfig initializes a Helm Configuration, ensures the use of a fresh configuration
+func (r *HelmClientIntitializer) InitializeHelmClientConfig() (*action.Configuration, error) {
+	settings := cli.New()
+	settings.KubeConfig = "/etc/kubernetes/admin.conf"
+
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(
+		settings.RESTClientGetter(),
+		settings.Namespace(),
+		os.Getenv("HELM_DRIVER"),
+		logAdapter,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize action config: %w", err)
+	}
+	return actionConfig, nil
+}
+
 func logAdapter(format string, v ...any) {
 	logrus.Debugf(format, v...)
 }
 
 // NewManager creates a new Component manager instance.
-func NewManager(snap snap.Snap) (*helmClient, error) {
+func NewManager(snap snap.Snap, initializers ...HelmConfigInitializer) (*helmClient, error) {
```

Comment:
This is also an option instead of passing `nil` default values, pick whichever you prefer. A simple `nil` arg is probably simpler until we have more configs to consider.

Reply:
I'm choosing the `nil` option for now but I will keep the functional pattern in mind if there is more configuration needed later.